aio/content/examples/schematics-for-libraries/projects/my-lib/schematics/my-service/index.ts

```diff
@@ -6,37 +6,51 @@ import {
   chain, mergeWith
 } from '@angular-devkit/schematics';
 
-import { strings, normalize, experimental } from '@angular-devkit/core';
+import { strings, normalize, virtualFs, workspaces } from '@angular-devkit/core';
 // #enddocregion schematics-imports
 
 import { Schema as MyServiceSchema } from './schema';
 // #enddocregion schema-imports
 
-export function myService(options: MyServiceSchema): Rule {
-  return (tree: Tree) => {
-    const workspaceConfig = tree.read('/angular.json');
-    if (!workspaceConfig) {
-      throw new SchematicsException('Could not find Angular workspace configuration');
-    }
+export function createHost(tree: Tree): workspaces.WorkspaceHost {
+  return {
+    async readFile(path: string): Promise<string> {
+      const data = tree.read(path);
+      if (!data) {
+        throw new SchematicsException('File not found.');
+      }
+      return virtualFs.fileBufferToString(data);
+    },
+    async writeFile(path: string, data: string): Promise<void> {
+      return tree.overwrite(path, data);
+    },
+    async isDirectory(path: string): Promise<boolean> {
+      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
+    },
+    async isFile(path: string): Promise<boolean> {
+      return tree.exists(path);
+    },
+  };
+}
 
-    // convert workspace to string
-    const workspaceContent = workspaceConfig.toString();
+export function myService(options: MyServiceSchema): Rule {
+  return async (tree: Tree) => {
+    const host = createHost(tree);
+    const { workspace } = await workspaces.readWorkspace('/', host);
 
-    // parse workspace string into JSON object
-    const workspace: experimental.workspace.WorkspaceSchema = JSON.parse(workspaceContent);
 // #enddocregion workspace
-// #docregion project-fallback
-    if (!options.project) {
-      options.project = workspace.defaultProject;
-    }
-// #enddocregion project-fallback
 
 // #docregion project-info
-    const projectName = options.project as string;
+    if (options.project === undefined) {
+      throw new SchematicsException('Option "project" is required.');
+    }
```

Comment:
Can the it to the following, to retain the previous behaviour.

```suggestion
    if (!options.project) {
      options.project = workspace.extensions.defaultProject;
    }
```